frontend/test/pytest/test_template.py

```diff
@@ -301,7 +448,47 @@ def kup(weights):
     params = jnp.array(np.random.random(size=(1, 6)))
     interpreted_fn = qml.QNode(kup, device)
     jitted_fn = qjit(interpreted_fn)
-    assert np.allclose(interpreted_fn(params), jitted_fn(params))
+
+    interpreted_result = interpreted_fn(params)
+    jitted_result = jitted_fn(params)
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_particle_conserving_u1(backend):
+    """Test particle conserving U1"""
+
+    def particle_conserving_u1(weights):
+        qml.ParticleConservingU1(weights, range(2), init_state=np.array([1, 1]))
+        return qml.expval(qml.PauliZ(0))
+
+    device = qml.device(backend, wires=2)
+    weights = jnp.array(np.random.random(size=(1, 1, 2)))
```

Comment:
We prefer not to put randomness in the tests. Could you choose some values? (this comment apply to other tests)